### ISSUE TITLE
[BUG] Build error on Respberry Pi 4B

### ISSUE THREAD

**Issue by super119** [MoAlyousef/fltk-rs#199]
If this is a known issue, please close it.

```
pi@raspberrypi:~/Projects/fltk-rs $ cargo build --release
    Updating crates.io index
   Compiling proc-macro2 v1.0.18
   Compiling unicode-xid v0.2.1
   Compiling cc v1.0.55
   Compiling syn v1.0.33
   Compiling cmake v0.1.44
   Compiling quote v1.0.7
   Compiling fltk-sys v0.7.4 (/home/pi/Projects/fltk-rs/fltk-sys)
   Compiling fltk-derive v0.7.4 (/home/pi/Projects/fltk-rs/fltk-derive)
   Compiling fltk v0.7.4 (/home/pi/Projects/fltk-rs)
error[E0308]: mismatched types
   --> src/text.rs:188:54
    |
188 |         unsafe { Fl_Text_Buffer_canUndo(self._inner, flag as i8) }
    |                                                      ^^^^^^^^^^ expected `u8`, found `i8`
    |
help: you can convert an `i8` to `u8` and panic if the converted value wouldn't fit
    |
188 |         unsafe { Fl_Text_Buffer_canUndo(self._inner, (flag as i8).try_into().unwrap()) }
    |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
  --> src/window.rs:89:54
   |
89 |         unsafe { Fl_Gl_Window_set_valid(self._inner, v as i8) }
   |                                                      ^^^^^^^ expected `u8`, found `i8`
   |
help: you can convert an `i8` to `u8` and panic if the converted value wouldn't fit
   |
89 |         unsafe { Fl_Gl_Window_set_valid(self._inner, (v as i8).try_into().unwrap()) }
   |                                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0308]: mismatched types
   --> src/window.rs:106:62
    |
106 |         unsafe { Fl_Gl_Window_set_context_valid(self._inner, v as i8) }
    |                                                              ^^^^^^^ expected `u8`, found `i8`
    |
help: you can convert an `i8` to `u8` and panic if the converted value wouldn't fit
    |
106 |         unsafe { Fl_Gl_Window_set_context_valid(self._inner, (v as i8).try_into().unwrap()) }
    |                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0308`.
error: could not compile `fltk`.

To learn more, run the command again with --verbose.
```

And a pretty straightforward change can fix it:

```
diff --git a/src/text.rs b/src/text.rs
index 2a556d893539..cfccdd04149e 100644
--- a/src/text.rs
+++ b/src/text.rs
@@ -185,7 +185,7 @@ impl TextBuffer {
     /// Sets whether the buffer can undo
     pub fn can_undo(&mut self, flag: bool) {
         assert!(!self._inner.is_null());
-        unsafe { Fl_Text_Buffer_canUndo(self._inner, flag as i8) }
+        unsafe { Fl_Text_Buffer_canUndo(self._inner, flag as u8) }
     }

     /// Loads a file into the buffer
diff --git a/src/window.rs b/src/window.rs
index ce485026af2f..0ead5ce50fd6 100644
--- a/src/window.rs
+++ b/src/window.rs
@@ -86,7 +86,7 @@ impl GlWindow {
     /// Mark the OpeGL context as still valid
     pub fn set_valid(&mut self, v: bool) {
         assert!(!self.was_deleted());
-        unsafe { Fl_Gl_Window_set_valid(self._inner, v as i8) }
+        unsafe { Fl_Gl_Window_set_valid(self._inner, v as u8) }
     }

     /// Returns whether the context is valid upon creation
@@ -103,7 +103,7 @@ impl GlWindow {
     /// Mark the context as valid upon creation
     pub fn set_context_valid(&mut self, v: bool) {
         assert!(!self.was_deleted());
-        unsafe { Fl_Gl_Window_set_context_valid(self._inner, v as i8) }
+        unsafe { Fl_Gl_Window_set_context_valid(self._inner, v as u8) }
     }

     /// Returns the GlContext
```

**Comment by MoAlyousef** [comment]
Hi
It should be a straightforward fix. I'll just change them to raw::c_char since u8/i8 is platform dependent.
I've pushed the changes to master. Will publish a patch this evening if all is green :)

**Comment by MoAlyousef** [comment]
fixed in #200
